server/client.go

```diff
@@ -4564,16 +4564,13 @@ func (c *client) processPingTimer() {
 	now := time.Now()
 	needRTT := c.rtt == 0 || now.Sub(c.rttStart) > DEFAULT_RTT_MEASUREMENT_INTERVAL
 
-	// Do not delay PINGs for GATEWAY or spoke LEAF connections.
-	if c.kind == GATEWAY || c.isSpokeLeafNode() {
+	// Do not delay PINGs for ROUTER, GATEWAY or spoke LEAF connections.
+	if c.kind == ROUTER || c.kind == GATEWAY || c.isSpokeLeafNode() {
 		sendPing = true
 	} else {
-		// If we have had activity within the PingInterval then
-		// there is no need to send a ping. This can be client data
-		// or if we received a ping from the other side.
-		if delta := now.Sub(c.last); delta < pingInterval && !needRTT {
-			c.Debugf("Delaying PING due to client activity %v ago", delta.Round(time.Second))
-		} else if delta := now.Sub(c.ping.last); delta < pingInterval && !needRTT {
+		// If we received a ping from the other side within the PingInterval then
+		// there is no need to send a ping.
+		if delta := now.Sub(c.ping.last); delta < pingInterval && !needRTT {
```

Comment:
This feels like it should be any inbound activity not just a ping.

Reply:
So other inbound data from a client connection will not suppress the pings here, that is the intended behavior correct?

Reply:
Well no. As per your suggestion on the other PR, in this update the c.ping.last timestamp is used to record both received pings and client data. There is another change to the end of the readLoop routine where c.ping.last is updated.

Reply:
ok maybe rename then to `in`, so c.in.last or c.inbound.last?

Reply:
I think we are close, and we are prepping to release 2.9.16. 

If we can get this merged later today or no later then tomorrow it will make it in.

Reply:
OK.  Not 100% sure this is what you meant, but I have changed the ping check to work with a `c.lastIn` timestamp, which is updated on receiving client data or a ping, and deleted the `c.ping.last` field.